File: solutions/cpp/perfect-numbers/3/perfect_numbers.cpp
#include "perfect_numbers.h"
#include <stdexcept>
namespace perfect_numbers {
    classification classify(long long value) {
        if (value < 1)
            throw std::domain_error("invalid value.");

        long long aliquot_sum{ 0 };
        for (long long n = 1; n <= value / 2; ++n) {
            if (value % n == 0)
                aliquot_sum += n;
        }

        if (value < aliquot_sum)
            return classification::abundant;
        else if (value > aliquot_sum)
            return classification::deficient;
        else
            return classification::perfect;
    }

}  // namespace perfect_numbers

File: solutions/python/perfect-numbers/2/perfect_numbers.py
CLASS_PERFECT: str = "perfect"
CLASS_ABUNDANT: str = "abundant"
CLASS_DEFICIENT: str = "deficient"


def classify(number: int) -> str:
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")

    aliquot_sum: int = 0
    for n in range(1, number // 2 + 1):
        aliquot_sum += n if number % n == 0 else 0

    return CLASS_PERFECT if aliquot_sum == number else (
        CLASS_ABUNDANT if aliquot_sum > number else CLASS_DEFICIENT)
